listeners/test-client.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+var Mocha = require('mocha');
+var fs = require('fs');
+var path = require('path');
+var Q = require('q');
+
+class TestClient {
+
+    constructor () {
+      var self = this;
+      self._testResult = Q.defer();
+      self._testName = Q.defer();
+    }
+
+    resetPromise(){
+      var self = this;
+      self._testResult = Q.defer();
+      self._testName = Q.defer();
+    }
+
+    getTestResult(){
+      var self = this;
+      return self._testResult.promise;
+    }
+
+    getTestName(){
+      var self = this;
+      return self._testName.promise;
+    }
+
+    runTest(asrResult, speakerIds){
+      var self = this;
+      var result = [];
+      var status = undefined;
+
+      //  Get the variables set by main.rule
+      var params = asrResult.NLParse;
+      if (params.action === 'testing') {
+          var dir = '';
+          switch (params.type) {
+              case 'unit':
+                  dir = './test/unit/';
+                  break;
+
+              case 'functional':
+                  dir = './test/func/';
+                  break;
+              default:
+                  status = {
+                    "status": false,
+                    "msg": "Can't run unknown tests!"
+                  };
+                  self._testName.resolve(status);
+                  break;
+          }
+          if (status == undefined){
```

Comment:
Instead of using an if check here, could we just exit when the switch default case happens?